providers/gitlab.go

```diff
@@ -331,7 +331,7 @@ func (p *GitLabProvider) addProjectsToSession(ctx context.Context, s *sessions.S
 				perms = projectInfo.Permissions.GroupAccess
 			}
 
-			if perms.AccessLevel >= project.AccessLevel {
+			if perms != nil && perms.AccessLevel >= project.AccessLevel {
```

Comment:
This mixed with the above `if perms == nil` fallback handling is slightly confusing. Can we check for `nil` on line 331 on `projectInfo.Permissions.GroupAccess` and `continue` if it is?

Can you also add a scenario for this case to the test table here: https://github.com/oauth2-proxy/oauth2-proxy/blob/4d9de06b1dbad8d27965afa2586df29667eeb50c/providers/gitlab_test.go#L189

Reply:
Thanks - I have reformatted it as suggested.